solr/solrj/src/java/org/apache/solr/client/solrj/impl/Http2SolrClient.java

```diff
@@ -533,6 +534,25 @@ public NamedList<Object> request(SolrRequest<?> solrRequest, String collection)
     }
   }
 
+  public <R> R requestWithBaseUrl(String baseUrl, SolrClientFunction<SolrClient, R> clientFunction)
```

Comment:
Conceptually, it probably makes sense for this method to be available on all of our "Http" clients.  But in practice, only Http2SolrClient makes it easy to quickly clone an existing instance, which this method implementation leans on pretty heavily.

Ambivalent on whether this is worth the effort to make available more widely.  Curious what others think...